soc/arm/st_stm32/stm32f1/Kconfig.defconfig.stm32f105xc

```diff
@@ -0,0 +1,25 @@
+# Kconfig - ST Microelectronics STM32F105XC MCU
+#
```

Comment:
Remove blank lines in Kconfig headers (recent common request).

Reply:
Copied the header format from `Kconfig.defconfig.stm32f107xc`